src/Core/Components/Progress/FluentProgressBar.razor.cs

```diff
@@ -0,0 +1,131 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Components;
+using Microsoft.FluentUI.AspNetCore.Components.Extensions;
+using Microsoft.FluentUI.AspNetCore.Components.Utilities;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// 
+/// </summary>
+public partial class FluentProgressBar : FluentComponentBase
+{
+    private readonly string _defaultId = Identifier.NewId();
+
+    /// <summary />
+    protected string? ClassValue => DefaultClassBuilder
+        .Build();
+
+    /// <summary />
+    protected string? StyleValue => DefaultStyleBuilder
+        .AddStyle("width", Width, () => !string.IsNullOrEmpty(Width))
+        .AddStyle("background-color", BackgroundColor, () => !string.IsNullOrEmpty(BackgroundColor))
+        .Build();
+
+    /// <summary>
+    /// Gets or sets the minimum value.
+    /// </summary>
+    [Parameter]
+    public int? Min { get; set; }
+
+    /// <summary>
+    /// Gets or sets the maximum value.
+    /// The FluentProgressBar bar will be full when value equals <see cref="Max"/>.
+    /// </summary>
+    [Parameter]
+    public int? Max { get; set; }
+
+    /// <summary>
+    /// Gets or sets the shape of the progress bar: rounded or square.
+    /// </summary>
+    [Parameter]
+    public ProgressShape? Shape { get; set; }
+
+    /// <summary>
+    /// Gets or sets the current value.
+    /// If `null` (default), the FluentProgressBar will display an indeterminate state.
+    /// </summary>
+    [Parameter]
+    public int? Value { get; set; }
+
+    /// <summary>
+    /// Gets or sets the visibility of the component
+    /// </summary>
+    [Parameter]
+    public bool Visible { get; set; } = true;
+
+    /// <summary>
+    /// Gets or sets the component width.
+    /// </summary>
+    [Parameter]
+    public string? Width { get; set; }
+
+    /// <summary>
+    /// Gets or sets the validation state of the progress bar: Success, Warning or Error.
+    /// </summary>
+    [Parameter]
+    public ProgressState? State { get; set; }
+
+    /// <summary>
+    /// Gets or sets the background color of the progress bar.
+    /// </summary>
+    [Parameter]
+    public string? BackgroundColor { get; set; }
+
+    /// <summary>
+    /// Gets or sets the color of the progress bar.
+    /// This property is not used when the <see cref="State"/> property is set.
+    /// </summary>
+    [Parameter]
+    public string? Color { get; set; }
+
+    /// <summary>
+    /// Gets or sets the stroke width of the progress bar.
+    /// If not set, the default theme stroke width is used.
+    /// </summary>
+    [Parameter]
+    [Obsolete("This property is not supported in FluentUI components and will be removed in a future release. Use Thickness property.")]
```

Comment:
[Obsolete("This property is not supported anymore and will be removed in a future release. Use Thickness property instead.")]